### PR TITLE
Issue 39 - Added Validate Parameter to xDNSServerAddress resource

### PR DESCRIPTION
**Pull request by PlagueHO**
Changes:
1. Added validate parameter to xDNSServerAddress resource. Defaults to $false.
2. Pester tests for Set-TargetResource updated mocks to detect if validate set or not.
3. Documentation and samples updated.

### CODE REVIEW

**Comment by PlagueHO** [comment]
Doh. Fixing conflicts.

File: DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1 [hunk 1]
```diff
@@ -54,6 +54,7 @@ function Get-TargetResource
             -AddressFamily $AddressFamily).ServerAddresses
         AddressFamily = $AddressFamily
         InterfaceAlias = $InterfaceAlias
+        Validate = $null
```
Comment (tysonjhayes) [inline DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1]: Do we have to do this? Just looking at it is making me cringe.

**Comment by tysonjhayes** [comment]
Besides my one comment it looks good to me.

File: DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1 (hunk 1 above)
Comment (PlagueHO) [inline DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1]: No, we definitely don't have to. :) All sorted now. :+1:

**Pull request merged**